roles/confluent.kafka_broker/templates/listener.j2

```diff
@@ -34,8 +37,22 @@ listener.name.{{listener.value.name|lower}}.gssapi.sasl.jaas.config=com.sun.secu
   keyTab="{{kerberos.keytab_dir}}/{{kafka_broker_kerberos_keytab_path | basename}}" \
   principal="{{kafka_broker_kerberos_principal}}";
 {% endif %}
+
 {% if listener['value']['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'SCRAM-SHA-256' %}
 listener.name.{{listener.value.name|lower}}.sasl.enabled.mechanisms=SCRAM-SHA-256
 listener.name.{{listener.value.name|lower}}.scram-sha-256.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required \
    username="{{sasl_scram_users.admin.principal}}" password="{{sasl_scram_users.admin.password}}";
 {% endif %}
+
+{% if listener['value']['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'OAUTHBEARER' %}
+listener.name.{{listener.value.name|lower}}.sasl.enabled.mechanisms=OAUTHBEARER
+listener.name.{{listener.value.name|lower}}.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+{% for option in listener.value.options|dict2items %}
+  {{ option['key'] }}="{{ option['value']}}"{% if loop.index != loop|length %} \
```

Comment:
Do you think the options list and server_callback and login_callback vars need to be vars here or can they be hardcoded?

Reply:
The base OAuthBearer config, when not being used for RBAC, has a bunch of options and support for your own callbacks. The intent was to keep the OAuthBearer listener definition generic for use outside of RBAC installations only. 

OAuthBearer Config/Options: https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_oauth.html

Reply:
@nerdynick OAuth bearer is not supported outside of the RBAC implementation, as its not a full Oauth mechanism. It cannot be used for authorizing clients for example.  I would recommend we don't include this at this time, until such time as RBAC is ready to be shipped.